### ISSUE TITLE
How tsconfig exclude pattern should actually work

### ISSUE THREAD

**Issue by whitecolor** [Microsoft/TypeScript#12824]
**TypeScript Version:**  2.1.1 / nightly (2.2.0-dev.201xxxxx)

Say we have a following structure to specially to test `exclude` pattern:

```
/node_modules
  /module/some.ts
nested/
  /node_modules
    /module/some.ts
  /deep-nested/
    /node_modules
      /module/some.ts
```
Will check this using `tsc --listFiles`

How *exclude* patterns work:
1) "node_modules" - will exclude **only** `/node_modules/module/some.ts`
2) "*/node_modules" (the same effect as */node_modules/*") - will exclude **only** `/nested/node_modules/module/some.ts`
3)  "**/node_modules" - will **exclude nothing**
4)  "**/node_modules/*" - will **exclude everything** (also nested, deep-nested)

Some behaviour seem strange to me 
- why in case 3 it excludes nothing (considering case 2 works)? 
- by default (if `exclude` not specified) case 1 will be effective, shouldn't it be 4) which excludes `node_modules` in nested folders as well?

**Comment by arusakov** [comment]
Related with #12750

**Comment by mhegazy** [comment]
looks like a duplicate of #12750

**Comment by whitecolor** [comment]
@mhegazy 
I posted there, also if first point (that `"**/node_modules"`doesn't work) is bug, there is second point about default exclusion pattern: currntly it is `"node_modules/*"` shouldn't it be "**/node_modules/*"?

**Comment by mhegazy** [comment]
> I posted there, also if first point (that "**/node_modules"doesn't work) is bug, there is second point about default exclusion pattern: currntly it is "node_modules/*" shouldn't it be "**/node_modules/*"?

possibly would consider it. the defaults are not meant to be comprehensive thought. it is just a starting point to get most users covered. you can always add additional ones by defining your own exclude patterns.

**Comment by whitecolor** [comment]
Yes but I think default should provide smart defaults, and I think if Ts so smart that event excludes three most common PM's fiolders (`node_modules`, `bower_components`, `jspm_modules`), it could also be smart enough to do it for nested folders as well.
